controllers/managedocs_controller.go

```diff
@@ -124,9 +152,24 @@ func (r *ManagedOCSReconciler) setDesiredStorageCluster(
 
 	// Handle strict mode reconciliation
 	if reconcileStrategy == v1.ReconcileStrategyStrict {
+		configSecret := &corev1.Secret{}
+		if err := r.Get(r.ctx, types.NamespacedName{Namespace: sc.ObjectMeta.Namespace, Name: r.ConfigSecretName}, configSecret); err != nil {
+			// Do not create the StorageCluster if the addon param secret is missing
+			r.Log.Info("Failed to get the addon param secret. Not reconciling the storage cluster.", "SecretName", r.ConfigSecretName)
+
+			return err
+		}
+		sdata := configSecret.Data
+
 		// Get an instance of the desired state
 		desired := utils.ObjectFromTemplate(templates.StorageClusterTemplate, r.Scheme).(*ocsv1.StorageCluster)
 
+		PersistentVolume := &desired.Spec.StorageDeviceSets[0].DataPVCTemplate
+		PersistentVolume.Spec.Resources = corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{
+				"storage": resource.MustParse(string(sdata[storageClassSizeKey])),
+			},
+		}
```

Comment:
The agreed way to have a specific size is by provisioning more 1TiB OSDS not by asking for a bigger device set. 
This also mandates the following missing checks:
1. Validate that the size is a multiply of 1TiB
2. Validate that the size is not already set in the actual storage cluster (we do not support resize at the moment)

Both conditions should log a message and terminate the reconcile loop with an  error

Reply:
I don't think we need the second check - the storageCluster cannot be created without a size

Reply:
the second check is to validate that the requested size is not different than what was set previously.  In that case we should abort

Reply:
In that case, should we just check if the storagecluster already exists and abort without checking for the secret?
If we cannot change the sc, there doesn't seem to be a point to watching the secret.

Reply:
The `CreateOrUpdate` callback does not and should not know if it exists or not. What we can do is check if the loaded resource has a size or not. Of course, this assumes that we do not have a default size in the template (which we should not).

This way it is easier to implement, more consistent with how reconciles are designed to work and, solves the problem at least as good as checking the existence of the storage cluster (I am aware that both methodologies are not perfect, but they have the same drawbacks)

This is also better groundwork  for the upcoming change that will enable us to support upscaling

Reply:
We could get the storagecluster object itself before calling CreateOrUpdate to check if it exists - if it does, don't do anything.

Reply:
Of course we can, but to me what you describe sounds backward.